apps/server/src/modules/tool/school-external-tool/uc/school-external-tool.uc.ts

```diff
@@ -40,15 +44,24 @@ export class SchoolExternalToolUc {
 		return createdSchoolExternalTool;
 	}
 
-	private async ensureSchoolPermission(userId: EntityId, schoolId: EntityId): Promise<void> {
-		return this.authorizationService.checkPermissionByReferences(userId, AuthorizableReferenceType.School, schoolId, {
-			action: Action.read,
-			requiredPermissions: [Permission.SCHOOL_TOOL_ADMIN],
-		});
+	private async ensureSchoolPermissions(
+		userId: EntityId,
+		tools: SchoolExternalTool[],
+		context: AuthorizationContext
+	): Promise<void> {
+		for (const tool of tools) {
```

Comment:
Please do not use awaits in for loops. You can achieve the same with Promise.all(tools.map( async () => doSomeStuff()));